Move d_do_test.d combination into a function.

marler8997 opened the pull request:
This is in preparation for #7099.  By moving the code for a single test combination into a function, the logic that drives all the combinations can be enhanced without having to increase the nesting of the combination code itself.

wilzbach reviewed (approved):
Looks great. Thanks!

wilzbach commented on test/d_do_test.d:
```diff
@@ -668,7 +669,7 @@ int tryMain(string[] args)
                 else version (linux)
                 {
-                    auto script = test_app_dmd_base ~ to!string(i) ~ ".gdb";
+                    auto script = test_app_dmd_base ~ to!string(permuteIndex) ~ ".gdb";
```
You need to update the `version(Windows)` block above too.

marler8997 replied:
done

wilzbach reviewed (dismissed):
Sorry - that was a bit hard to catch.

wilzbach reviewed (approved):
I will wait for another pairs of eyes then.

Commit:
Move d_do_test.d combination into a function.

This is in preparation for PR 7066.  By moving the code for a single test combination into a function, the logic that drives all the combinations can be enhanced without having to increase the nesting of the combination code itself.

stefan-koch-sociomantic commented:
that code is rather unclear due to the lack of documentation. 
If a function is doing something that is not trivial please share it in a sentence or two.

marler8997 commented:
Ok I added a comment for the `testCombination` function.  Since this PR is fairly trivial and blocking #7099, could we get this merged sooner rather than later?
